src/coreclr/jit/gentree.cpp

```diff
@@ -4205,7 +4205,7 @@ unsigned Compiler::gtSetEvalOrder(GenTree* tree)
     DONE_OP1_AFTER_COST:
 
         bool bReverseInAssignment = false;
-        if (oper == GT_ASG)
+        if (oper == GT_ASG && (!optValnumCSE_phase || optCSE_canSwap(op1, op2)))
```

Comment:
I see that you also use optCSE_canSwap,  that allows us to reorder more cases and should result in the minimal diffs.
Probably only the broken cases would have diffs.